test/Quest1155.t.sol

```diff
@@ -0,0 +1,306 @@
+
+        vm.prank(questFactoryMock);
+        quest.queue();
+        assertTrue(quest.queued(), "queued should be true");
+    }
+
+
+    /*//////////////////////////////////////////////////////////////
+                            SINGLECLAIM
+    //////////////////////////////////////////////////////////////*/
+
+    function test_singleClaim() public {
+        SampleERC1155(sampleERC1155).mintSingle(address(quest), 1, 100000);
+        vm.deal(address(quest), 100000000);
+        vm.prank(questFactoryMock);
+        quest.queue();
+
+        // emit ClaimedSingle(account_, rewardToken, 1);
+        vm.expectEmit(true, true, false, false, address(quest));
+        emit ClaimedSingle(participant, address(sampleERC1155), TOKEN_ID);
+
+        uint256 protocolFeeRecipientOGBalance = protocolFeeRecipient.balance;
+        vm.prank(questFactoryMock);
+        quest.singleClaim(participant);
+
+        assertEq(
+            protocolFeeRecipient.balance,
+            protocolFeeRecipientOGBalance + QUEST_FEE,
+            "participant should have received the reward"
+        );
+        assertEq(
+            SampleERC1155(sampleERC1155).balanceOf(participant, TOKEN_ID),
+            1,
+            "participant should have received the reward"
+        );
+    }
+
+    // todo add fuzz test
+
+    function test_RevertIf_singleClaim_NotQuestFactory() public {
+        SampleERC1155(sampleERC1155).mintSingle(address(quest), 1, 100000);
+        vm.deal(address(quest), 100000000);
+        vm.prank(questFactoryMock);
+        quest.queue();
+
+        vm.expectRevert(abi.encodeWithSelector(NotQuestFactory.selector));
+        quest.singleClaim(participant);
+    }
+
+    function test_RevertIf_singleClaim_NotStarted() public {
+        SampleERC1155(sampleERC1155).mintSingle(address(quest), 1, 100000);
+        vm.deal(address(quest), 100000000);
+        vm.prank(questFactoryMock);
+        quest.queue();
+
+        vm.warp(START_TIME - 1);
+        vm.prank(questFactoryMock);
+        vm.expectRevert(abi.encodeWithSelector(NotStarted.selector));
+        quest.singleClaim(participant);
+    }
+
+    function test_RevertIf_singleClaim_whenNotPaused() public {
+        SampleERC1155(sampleERC1155).mintSingle(address(quest), 1, 100000);
+        vm.deal(address(quest), 100000000);
+        vm.prank(questFactoryMock);
+        quest.queue();
+        vm.startPrank(questFactoryMock);
+        quest.pause();
+        vm.warp(START_TIME);
+        vm.expectRevert("Pausable: paused");
+        quest.singleClaim(participant);
+        vm.stopPrank();
+    }
+
+    // /*//////////////////////////////////////////////////////////////
+    //                   WITHDRAWREMAININGTOKENS
+    // //////////////////////////////////////////////////////////////*/
+
+    function test_withdrawRemainingTokens() public {
+        SampleERC1155(sampleERC1155).mintSingle(address(quest), 1, 100000);
+        vm.deal(address(quest), 100000000);
```

Comment:
If possible it'd be good to use a const vs a magic number here. Makes changing the test variable in the future easier and makes it a lot lower cognitive load to create the fuzz tests since everything is already setup deterministically.